packages/thirdweb/src/chainsaw/formatter.ts

```diff
@@ -0,0 +1,132 @@
+import { formatBlock, formatTransaction } from "viem";
+import { numberToHex } from "../utils/encoding/hex.js";
+import { parseNFT } from "../utils/nft/parseNft.js";
+import type {
+  Block,
+  ChainsawBlock,
+  ChainsawEvents,
+  ChainsawNFTs,
+  ChainsawTransaction,
+  ChainsawTransactions,
+  Events,
+  NFTs,
+  Transactions,
+} from "./types.js";
+
+export function formatChainsawBlock(block?: Block): ChainsawBlock | undefined {
+  if (!block) return;
+  return formatBlock({
+    ...block,
+    baseFeePerGas: block.baseFeePerGas
+      ? numberToHex(BigInt(block.baseFeePerGas))
+      : undefined,
+    difficulty: numberToHex(BigInt(block.difficulty)),
+    gasLimit: numberToHex(BigInt(block.gasLimit)),
+    gasUsed: numberToHex(BigInt(block.gasUsed)),
+  });
+}
+
+export function formatChainsawNFTs(nfts?: NFTs): ChainsawNFTs {
+  if (!nfts) return [];
+  return nfts.map((nft) => {
+    const options =
```

Comment:
Up to you (don't have enough visibility into chainsaw to know) but it could be good to make this a switch statement so we make sure to cover all possible options for `nft.type` (in case we add labels for things like ERC721C in the future)

Reply:
Done in [3ab5655](https://github.com/thirdweb-dev/js/pull/4004/commits/3ab5655cff7a4d54acb7eeff67391b1855eef395)